### PR TITLE
[DROOLS-6048] Enforcing before rules/after rules semantic

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @mariofusco @jiripetrlik

See https://issues.redhat.com/browse/DROOLS-6048

Scope of this PR is to enforce/clarify the "before rules/after rules" semantic introduced by DROOLS-5950.
In detail:

1. renamed some methods for consistency
2. uniformed methods
3. replaced addResource/addResources with addResourceAfterRules / addResourcesAfterRules

@tarilabs @evacchi FYI

Depends on https://github.com/kiegroup/droolsjbpm-knowledge/pull/503

### CODE REVIEW

**Comment by tarilabs** [comment]
Thank you for the FYI.
I didn't understand if the test which you mentioned previously error-ed on your work-in-progress, now are again green after the full set of code change?

**Comment by gitgabrio** [comment]
Jenkins run cdb

**Comment by gitgabrio** [comment]
> Thank you for the FYI.
> I didn't understand if the test which you mentioned previously error-ed on your work-in-progress, now are again green after the full set of code change?

Looking at the code I had I could not find a reason for that error. I tried to debug a little, but since it was taking too much time,  I've restarted such renaming from scratch - with a slightly different approach - and the error is gone.

**Comment by tarilabs** [comment]
> I've restarted such renaming from scratch - with a slightly different approach - and the error is gone.

👍  glad to hear you achieved the desired result, possibly it was just the IDE missed some update during the 1st round of refactoring.

**Review by danielezonca — APPROVED** [review]

File: drools-compiler/src/main/java/org/drools/compiler/builder/impl/CompositeKnowledgeBuilderImpl.java [hunk 1]
```diff
@@ -215,6 +202,19 @@ private void buildAssemblerResourcesBeforeRules() {
...
     }
 
+    private void buildAssemblerResourcesAfterRules() {
+        KieAssemblers assemblers = ServiceRegistry.getService(KieAssemblers.class);
+        try {
+            for (Map.Entry<ResourceType, List<ResourceDescr>> entry : resourcesByType.entrySet()) {
+                List<ResourceWithConfiguration> rds = entry.getValue().stream().map(CompositeKnowledgeBuilderImpl::descrToResourceWithConfiguration).collect(Collectors.toList());
+                assemblers.addResourcesAfterRules(kBuilder, rds, entry.getKey());
+            }
+        } catch (RuntimeException e) {
```
Comment (jiripetrlik) [inline drools-compiler/src/main/java/org/drools/compiler/builder/impl/CompositeKnowledgeBuilderImpl.java:212]: This looks quite unusual to me. Can we for example unify both catch blocks and always throw Runtime exception with reasonable description and original exception inside? Description might be something like "Building assembler resources failed".

**Review by jiripetrlik — COMMENTED** [review]
Looks good. I have one suggestion.

File: drools-compiler/src/main/java/org/drools/compiler/builder/impl/CompositeKnowledgeBuilderImpl.java (hunk 1 above)
Comment (gitgabrio) [reply drools-compiler/src/main/java/org/drools/compiler/builder/impl/CompositeKnowledgeBuilderImpl.java:212]: @jiripetrlik 
Hi! Scope of this PR is just plain renaming for sake of improved readability.
Behind those two different exception management (buildAssemblerResourcesBeforeRules and buildAssemblerResourcesAfterRules) there is the way how the whole flow is managed and the exception are used (pls look inside CompositeKnowledgeBuilderImpl.build(boolean buildRules))
I can't exactly follow why they are so different, but it is a little bit out of scope from this PR.

**Comment by gitgabrio** [comment]
@danielezonca Hi! Could you please merge this, squashing commits? Thanks!

**Pull request merged**